lib/widgets/topic_list.dart

```diff
@@ -149,31 +147,30 @@ class _TopicListState extends State<_TopicList> with PerAccountStoreAwareStateMi
     });
   }
 
-  void _fetchTopics() async {
+  void _fetchTopics(PerAccountStore store) async {
     // Do nothing when the fetch fails; the topic-list will stay on
     // the loading screen, until the user navigates away and back.
     // TODO(design) show a nice error message on screen when this fails
-    final store = PerAccountStoreWidget.of(context);
-    final result = await getStreamTopics(store.connection,
-      streamId: widget.streamId,
-      allowEmptyTopicName: true);
+    await store.fetchTopics(widget.streamId);
     if (!mounted) return;
     setState(() {
-      lastFetchedTopics = result.topics;
+      // The actual state lives in the PerAccountStore.
     });
   }
 
   @override
   Widget build(BuildContext context) {
-    if (lastFetchedTopics == null) {
+    final store = PerAccountStoreWidget.of(context);
+    final channelTopics = store.getChannelTopics(widget.streamId);
+    if (channelTopics == null) {
       return const Center(child: CircularProgressIndicator());
     }
 
     // TODO(design) handle the rare case when `lastFetchedTopics` is empty
```

Comment:
nit:

```dart
    // TODO(design) handle the rare case when `channelTopics` is empty
```

Would you check if there's an issue for this TODO to refer to, and file a brief issue if not?